rpmlint/permissions.py

```diff
@@ -0,0 +1,171 @@
+            # package is not yet around
+            pass
+
+    def _parse(self, label, fd):
+        for nr, line in enumerate(fd.readlines(), 1):
+            line = line.strip()
+
+            if not line or line.startswith('#'):
+                continue
+
+            parts = line.split('=', 1)
+
+            if len(parts) != 2:
+                raise Exception(f'{label}:{nr}: parse error')
+
+            varname = parts[0].strip()
+            values = parts[1].split()
+            # strip leading or trailing slashes
+            values = [v.strip(os.path.sep) for v in values]
+
+            self.variables[varname] = values
+
+    def expandPaths(self, path):
+        """Checks for %{...} variables in the given path and expands them, as
+        necessary. Will return a list of expanded paths, will be only a single
+        path if no variables are used."""
+
+        ret = ['']
+
+        for part in path.split(os.path.sep):
+            if part.startswith('%{') and part.endswith('}'):
+                # variable found
+                variable = part[2:-1]
+                try:
+                    expansions = self.variables[variable]
+                except KeyError:
+                    raise Exception(f"Undeclared variable '{variable}' encountered in profile")
+
+                new_ret = []
+                for p in ret:
+                    for value in expansions:
+                        new_ret.append(os.path.sep.join([p, value]))
+
+                ret = new_ret
+            elif not part:
+                # a leading slash, ignore
+                continue
+            else:
+                # a regular, fixed string
+                ret = [os.path.sep.join([p, part]) for p in ret]
+
+        if path.endswith(os.path.sep):
+            # restore trailing slashes since they signify that we
+            # expect a directory
+            ret = [p + os.path.sep for p in ret]
+
+        return ret
+
+
+class PermissionsParser:
+
+    def __init__(self, var_handler, profile_path):
+        self.var_handler = var_handler
+        self.entries = {}
+
+        with open(profile_path) as fd:
+            self._parseFile(profile_path, fd)
+
+    def _parseFile(self, _label, fd):
+        context = ParseContext(_label)
+        for nr, line in enumerate(fd.readlines(), 1):
+            line = line.strip()
+            if not line or line.startswith('#'):
+                continue
+
+            context.line_nr = nr
+            self._parseLine(context, line)
+
+    def _parseLine(self, context, line):
+        if line.startswith('/') or line.startswith('%'):
+            context.active_entries = []
+
+            entry = PermissionsEntry(context.label, context.line_nr)
+            path, ownership, mode = line.split()
+            # the format supports both "user.group" and
+            # "user:group"
+            entry.owner, entry.group = ownership.replace('.', ':').split(':')
+            entry.mode = int(mode, 8)
```

Comment:
What about doing that in constructor?

Reply:
Sorry, I don't understand it, do you mean removing `_parse_file` and `_parse_line` and putting everything in the constructor?